src/main/java/tigase/xmpp/impl/PresenceAbstract.java

```diff
@@ -520,7 +523,7 @@ public void broadcastProbe(XMPPResourceConnection session, Queue<Packet> results
 					if (log.isLoggable(Level.FINEST)) {
 						log.log(Level.FINEST, session.getBareJID() + " | Sending initial presence to: " + buddy);
 					}
-					sendPresence(null, null, buddy, results, presInit);
+					sendPresence(null, session.getBareJID(), buddy.getBareJID(), results, presInit);
```

Comment:
Why do we assume that `DynamicRoster` cannot have items with full jid?

Reply:
Also, I do not thing that we should send `from` bare jid, however as `sendPresence` uses `from` only if `presence` is not set, then it could work.

Reply:
I'm not sure why RFC6121 states to use bare jid for presence probes if the roster can keep full jids. In this case, should we still send probes to bare jid or rather to the full jid?

Reply:
I think that FullJIDs in roster is something rather obscure (and doesn't make all that much sense) so I think it should be ok to use BareJID here. If you can think of a good example were using FullJID here would be required then we can switch to FullJIDs but otherwise I would lean towards keeping BareJIDs.

Reply:
I do not recall any active usage of that, so I assume this change would be OK. Especially, since this is for the presence probe.